reset.c

```diff
@@ -0,0 +1,593 @@
+                return EXIT_FAILURE;
+            }
+            resetconf.pgname = optarg;
+        }
+        break;
+        case 'o':                           //output file name
+        {
+            if (outname) {                  //already given!
+                usage(stderr);
+                return EXIT_FAILURE;
+
+            }
+            outname = optarg;
+        }
+        break;
+        case 'x':                           //remove aux tag
+        {
+            if (*optarg == '^') {           //remove all except given ones!
+                if (parse_aux_list(&resetconf.aux_keep, optarg+1, "main_reset")) {
+                    usage(stderr);
+                    return EXIT_FAILURE;
+                }
+            }
+            else {                          //remove given ones
+                if (parse_aux_list(&resetconf.aux_remove, optarg, "main_reset")) {
+                    usage(stderr);
+                    return EXIT_FAILURE;
+                }
+            }
+        }
+        break;
+        case LONG_OPT('x'):                  //keep aux tags
+        {
+            if (parse_aux_list(&resetconf.aux_keep, optarg, "main_reset")) {
+                usage(stderr);
+                return EXIT_FAILURE;
+            }
+        }
+        break;
+        case '@':                           //thread count
+        {
+            if (ga.nthreads != 0) {
+                usage(stderr);
+                return EXIT_FAILURE;
+            }
+            ga.nthreads = atoi(optarg);
+        }
+        break;
+        // handle standard samtool options like thread count, verbosity...
+        default:
+            if (parse_sam_global_opt(c, optarg, lopts, &ga) == 0)
+                break;
+            // else fall-through
+        // couldn't parse or unknown options, show usage!
+        case '?':                           //unknown options found!
+        {
+            usage(stderr);
+            return EXIT_FAILURE;
+            break;
+        }
+
+        }
+    }
+
+    if (argc == 1 && isatty(STDIN_FILENO)) {
+        //no args and input is stdin -- it is the usage check
+        usage(stdout);
+        return EXIT_SUCCESS;
+    }
+    //else have other args or input from redirection/pipe/other device -- validate and work
+
+    if (!outname)
+        outname = "-";
+
+    //check and fail if unnecessary parameters are given
+    c = argc - optind;
+    if (c > 1) {
+        usage(stderr);
+        return EXIT_FAILURE;
+    }
+
+    if (c == 1) {
+        inname = argv[optind];
+    }
+    else {
+        inname = "-";
+    }
+
+    //set output file format based on name
+    sam_open_mode(outmode + 1, outname, NULL);
+
+    //open input and output files
+    infile = sam_open(inname, "r");
+    outfile = sam_open_format(outname, outmode, &ga.out);
+    if (!infile || !outfile) {
+        fprintf(stderr, "Could not open %s%s%s", !infile ? inname : "", (!infile && !outfile)? ", " : "", !outfile ? outname : "");
```

Comment:
Missing newline on the end of the fprintf statement.

Reply:
Done.